crud/select/compat/select.lua

```diff
@@ -23,6 +23,7 @@ local function build_select_iterator(space_name, user_conditions, opts)
         first = '?number',
         batch_size = '?number',
         bucket_id = '?number|cdata',
+        bucket_optimization = '?boolean',
```

Comment:
I don't like this option name.
I would like it to be "negative". Like "don't use optimization".  @olegrok, what do you think?

But if you left this flag "positive", you can simply use `opts.bucket_optimization != false` instead of setting it to `true` if it's `nil`.

Reply:
To be honest I also don't like this option name. Also amount of already existed options makes me frustrating.

I suggest to reconsider approach to solution of this problem.

My thoughts:
  * Add option that states that following call will we a map call (I also don't like it. It's still one more option and it looks like a crutch).
  * Take 2 steps back to take a step forward. Initially such optimization protects user from redundant network hops if it's possible to calculate bucket_id from user defined filters. If user uses custom sharding it would be great to consider it - we need to provide a way to change rules of calculation of bucket_id. It's a bit complex and ambitious feature but I like it.
  * Allow to put "box.NULL" to bucket_id option to block bucket_id calculation (It makes `box.NULL` a kind system/special value. And I also hates such approaches. But it could be solution).